src/persist-client/tests/trace/compaction_apply_res

```diff
@@ -0,0 +1,133 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+push-batch
+[0][2][0] 1 k0
+[2][4][0] 1 k1
+[4][6][0] 1 k2
+[6][8][0] 1 k3
+[8][10][0] 1 k4
+[10][12][0] 1 k5
+[12][14][0] 1 k6
+[14][16][0] 1 k7
+----
+ok
+
+push-batch
+[16][18][0] 100 k8
+----
+ok
+
+spine-batches
+----
+[0][16][0] 8/8 k0 k1 k2 k3 k4 k5 k6 k7
+[16][18][0] 100 k8
+
+# cannot apply batch that spans two spine batches exactly
+apply-merge-res
+[0][18][0] 2 nope
+----
+no-op
+
+# cannot apply batch that spans two spine batches with matching lower
+apply-merge-res
+[0][17][0] 2 nope
+----
+no-op
+
+# cannot apply batch that spans two spine batches with matching upper
+apply-merge-res
+[1][18][0] 2 nope
+----
+no-op
+
+# cannot apply batch that spans more than all spine batches
+apply-merge-res
+[0][20][0] 2 nope
+----
+no-op
+
+# cannot apply batch whose lower does not align with the lower of any parts of our fueled merge
+apply-merge-res
+[1][6][0] 2 nope
+----
+no-op
+
+# cannot apply batch whose upper does not align with the upper of any parts of our fueled merge
+apply-merge-res
+[2][7][0] 2 nope
+----
+no-op
+
+# cannot apply batch whose lower nor upper aligns with any parts of our fueled merge
+apply-merge-res
+[2][7][0] 2 nope
+----
+no-op
+
+# cannot apply batch into a merge spine batch
+apply-merge-res
+[16][17][0] 2 nope
+----
+no-op
+
+# perform a merge res at the start of the fueled merge
+apply-merge-res
+[0][4][0] 2 k0-1
+----
+applied subset
+
+spine-batches
+----
+[0][16][0] 7/8 k0-1 k2 k3 k4 k5 k6 k7
+[16][18][0] 100 k8
+
+# perform a merge res at the end of the fueled merge
+apply-merge-res
+[12][16][0] 2 k6-7
+----
+applied subset
+
+spine-batches
+----
+[0][16][0] 6/8 k0-1 k2 k3 k4 k5 k6-7
+[16][18][0] 100 k8
+
+# perform a merge res in the middle of the fueled merge
+apply-merge-res
+[6][10][0] 2 k3-4
+----
+applied subset
+
+spine-batches
+----
+[0][16][0] 5/8 k0-1 k2 k3-4 k5 k6-7
+[16][18][0] 100 k8
+
+# replace a single part within the fueled merge
+apply-merge-res
+[4][6][0] 1 k2_2
+----
+applied subset
+
+spine-batches
+----
+[0][16][0] 5/8 k0-1 k2_2 k3-4 k5 k6-7
+[16][18][0] 100 k8
+
+# perform a merge res that replaces an entire spine batch
```

Comment:
one last test case for you. replacing an already replaced batch plus some other stuff (simulating the back half of start 1-3, start 1-5, apply 1-3, apply 1-5). so like swap out `k3-4 + k5` here or something